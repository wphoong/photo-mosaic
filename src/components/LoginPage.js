import React from "react";
import { connect } from "react-redux";
import { startLogin } from "../actions/auth.js";

console.log("connected to login page");

export const LoginPage = ({startLogin}) => (
  <div className="box-layout" >
    <div className="box-layout__box">
      <h1 className="box-layout__title">Photo Mosaic</h1>
      <p>Enjoy your photos anytime!</p>
      <button className="button" onClick={startLogin}>Login with Google</button>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);