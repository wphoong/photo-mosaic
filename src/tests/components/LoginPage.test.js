import React from "react";
import { shallow } from "enzyme";
import { LoginPage } from "../../components/LoginPage.js";

test("should render Login page", () => {
  const wrapper = shallow(<LoginPage />);
  expect(wrapper).toMatchSnapshot();
});

test("should render Login page", () => {
  const startLogInSpy = jest.fn();
  const wrapper = shallow(<LoginPage startLogin={startLogInSpy} />);
  wrapper.find("button").simulate("click");
  expect(startLogInSpy).toHaveBeenCalled();
});