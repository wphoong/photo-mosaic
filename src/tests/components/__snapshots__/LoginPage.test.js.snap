// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Login page 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Photo Mosaic
    </h1>
    <p>
      Enjoy your photos anytime!
    </p>
    <br />
    <button
      className="button"
    >
      Login with Google
    </button>
  </div>
</div>
`;
